src/index.d.ts

```diff
@@ -1,29 +1,112 @@
-export type Version = 0 | 1
-export type Codec = string
-export type Multihash = Buffer
-export type BaseEncodedString = string
-export type MultibaseName = string
-
-export class CID {
-  constructor(version: Version, codec: Codec, multhash: Multihash, multibaseName?: MultibaseName)
-  constructor(cidStr: BaseEncodedString)
+/**
+ * Class representing a CID `<mbase><version><mcodec><mhash>`
+ * , as defined in [ipld/cid](https://github.com/multiformats/cid).
+ */
+declare class CID {
+  /**
+   * Create a new CID.
+   *
+   * The algorithm for argument input is roughly:
+   * ```
+   * if (cid)
+   *   -> create a copy
+   * else if (str)
+   *   if (1st char is on multibase table) -> CID String
+   *   else -> bs58 encoded multihash
+   * else if (Buffer)
+   *   if (1st byte is 0 or 1) -> CID
+   *   else -> multihash
+   * else if (Number)
+   *   -> construct CID by parts
+   * ```
+   *
+   * @example
+   * new CID(<version>, <codec>, <multihash>, <multibaseName>)
+   * new CID(<cidStr>)
+   * new CID(<cid.buffer>)
+   * new CID(<multihash>)
+   * new CID(<bs58 encoded multihash>)
+   * new CID(<cid>)
+   */
+  constructor(version: 0 | 1, codec: string, multhash: Buffer, multibaseName?: string)
+  constructor(cidStr: string)
   constructor(cidBuf: Buffer)
-  constructor(cidMultihash: Multihash)
+  constructor(cidMultihash: Buffer)
   constructor(cid: CID)
```

Comment:
really minor niggle - I find these easier to review if the overrides are listed in implementation order, so moving this one up would help, I guess having the full one listed first isn't such a bad thing

Reply:
No probs, that's a good/valid request. I'll update.